libkbfs/folder_block_ops.go

```diff
@@ -1042,27 +1060,76 @@ func (fbo *folderBlockOps) pathFromNodeForBlockWriteLocked(
 	return p, nil
 }
 
+// writeGetFileLocked checks write permissions explicitly for
+// writeDataLocked, truncateLocked etc and returns
+func (fbo *folderBlockOps) writeGetFileLocked(
+	ctx context.Context, lState *lockState, md *RootMetadata,
+	file path) (*FileBlock, keybase1.UID, error) {
+	fbo.blockLock.AssertLocked(lState)
+
+	username, uid, err := fbo.config.KBPKI().GetCurrentUserInfo(ctx)
+	if err != nil {
+		return nil, "", err
+	}
+	if !md.GetTlfHandle().IsWriter(uid) {
+		return nil, "", NewWriteAccessError(md.GetTlfHandle(), username)
+	}
+	fblock, err := fbo.getFileLocked(ctx, lState, md, file, blockWrite)
+	if err != nil {
+		return nil, "", err
+	}
+	return fblock, uid, nil
+}
+
+// createIndirectBlockLocked creates a new indirect block and
+// pick a new id for the existing block, and use the existing block's ID for
+// the new indirect block that becomes the parent.
+func (fbo *folderBlockOps) createIndirectBlockLocked(md *RootMetadata,
+	file path, uid keybase1.UID, dver DataVer) (*FileBlock, error) {
+
+	newID, err := fbo.config.Crypto().MakeTemporaryBlockID()
+	if err != nil {
+		return nil, err
+	}
+	fblock := &FileBlock{
+		CommonBlock: CommonBlock{
+			IsInd: true,
+		},
+		IPtrs: []IndirectFilePtr{
+			{
+				BlockInfo: BlockInfo{
+					BlockPointer: BlockPointer{
+						ID:       newID,
+						KeyGen:   md.LatestKeyGeneration(),
+						DataVer:  dver,
```

Comment:
I believe this will just get overridden at https://github.com/keybase/kbfs/blob/master/libkbfs/folder_block_ops.go#L1734, in the `ReadyBlock` call.  It's probably best to just set the data version during `ReadyBlock` somehow, on the top-level indirect block if possible.